### ISSUE TITLE
Infinite recursion when exporting glTF, serializing a Mesh

### ISSUE THREAD

**Issue by garyo** [arose/ngl#580]
I'm trying to export a scene from ngl as glTF. The exporter at some point calls `JSON.stringify( object.userData )` on a Mesh object, uuid CC31D3EB.
* which calls JSON.stringify(object.userData): userData here is an object containing a  buffer: CylinderGeometryBuffer and an instance (undefined).
* which calls toJSON on a BufferGeometry (I think contained within the CylinderGeometryBuffer), uuid=0fb349…, and that returns OK
* the next call is on a Group, uuid=0E198257 -- this is the group inside the Mesh, as in `Mesh.userData.buffer.group`
  * It stringifies the group's userData, but it's empty so does nothing.
* But: the group has 1 child, which now gets toJSON called on it; the child is a Mesh, uuid=CC31D3EB: AHA, the same mesh!
So at that point it recurses infinitely.

Basically, the problem boils down to this: `object.userData.buffer.group.children[0] === object` on the Mesh object.

I recognize this could just as easily be a three.js bug as an ngl bug, so I'm hoping for your help in tracking it down.

If the group in there refers to the same mesh, I don't see how to avoid a loop. Perhaps a custom `toJSON` method in the right place could help? I'm happy to contribute a fix if someone can point me in the right direction. I'm guessing it should somehow avoid serializing the buffer's `group` member in this case?

**Comment by garyo** [comment]
See my PR #581

**Comment by garyo** [comment]
This is now fixed by #609 and #663.
